### PR TITLE
Feature/parameterserver

### PR DESCRIPTION
**Pull request by rickaha**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
ParameterServer can only be used on vehicle

* **What is the new behavior (if this is a feature change)?**
ParameterServer can also be used on ControlTower

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:
This is a needed feat when using for example waypointfollower or followpoint over remote connection. You can provide parameters to the UI and get/set them

### CODE REVIEW

**Review by chrpro — CHANGES_REQUESTED** [review]
Looks good, just two minor improvements

File: communication/parameterserver.cpp [hunk 1]
```diff
@@ -22,3 +32,33 @@ void ParameterServer::updateParameter(std::string parameterName, float parameter
...
+    const std::lock_guard<std::mutex> lock(mMutex);
+    mParameterToClassMapping.insert_or_assign(parameterName, std::make_pair(setClassParameterFunction, getClassParameterFunction));
+};
+
+void ParameterServer::saveParametersToXmlFile(QString filename)
+{
+    const std::lock_guard<std::mutex> lock(mMutex);
+    QFile parameterFile(filename);
+
+    if (!parameterFile.open(QIODevice::WriteOnly))
```
[the remark this answers is not in the document]
Comment (rickaha) [reply communication/parameterserver.cpp]: Good catch! Added a return

File: communication/parameterserver.cpp [hunk 2]
```diff
@@ -26,11 +26,15 @@ ParameterServer* ParameterServer::getInstance()
     return mInstancePtr;
 }
 
-void ParameterServer::updateParameter(std::string parameterName, float parameterValue)
+bool ParameterServer::updateParameter(std::string parameterName, float parameterValue)
 {
     const std::lock_guard<std::mutex> lock(mMutex);
-    auto setParameterFunction = mParameterToClassMapping.find(parameterName)->second.first;
-    setParameterFunction(parameterValue);
+    if (auto search = mParameterToClassMapping.find(parameterName); search != mParameterToClassMapping.end()) {
```
[the remark this answers is not in the document]
Comment (rickaha) [reply communication/parameterserver.cpp]: Done

**Review by chrpro — APPROVED** [review]
looks good

**Pull request merged**